### PR TITLE
[Enhancement] Fix some hudi cve problems

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
Bump hudi from 0.14.1 -> 0.15.0, it fixed some cve problems(hbase's dependency still can't removed).
Exclude zookeeper, so we don't need to `rm -r zookeeper` in build.sh

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -549,24 +549,10 @@ if [ ${BUILD_BE} -eq 1 ]; then
     cp -p ${STARROCKS_THIRDPARTY}/installed/gcs_connector/*.jar ${STARROCKS_OUTPUT}/be/lib/hadoop/hdfs
     cp -r -p ${STARROCKS_THIRDPARTY}/installed/hadoop/lib/native ${STARROCKS_OUTPUT}/be/lib/hadoop/
 
-    # remove log4j
-    rm -f ${STARROCKS_OUTPUT}/be/lib/hadoop/common/lib/log4j-1.2.17.jar
```
Comment (Smith-Cruise) [inline build.sh:553]: already not existed

File: fe/fe-core/src/main/java/com/starrocks/connector/hudi/HudiRemoteFileIO.java [hunk 2]
```diff
@@ -49,10 +51,10 @@
 
 public class HudiRemoteFileIO implements RemoteFileIO {
     private static final Logger LOG = LogManager.getLogger(HudiRemoteFileIO.class);
-    private final Configuration configuration;
+    private final HadoopStorageConfiguration configuration;
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/hudi/HudiRemoteFileIO.java:54]: hudi changed it's api in 0.15

File: fe/fe-core/pom.xml [hunk 3]
```diff
@@ -543,6 +543,10 @@ under the License.
                     <groupId>org.apache.hadoop</groupId>
                     <artifactId>hadoop-client-runtime</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>org.apache.commons</groupId>
```
[the remark this answers is not in the document]
Comment (Smith-Cruise) [reply fe/fe-core/pom.xml:547]: spark_dpp is using `commons-compress-1.22.0`, it contains cve problems. And we already have `commons-compress-1.24.0` from `hadoop-common-3.4.0`

**Review by stephen-shelby — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java
Comment (plotor) [inline fe/fe-core/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java:1]: @Smith-Cruise why remove this file, I found that the `NoSuchMethodError` problem described in [#46886](https://github.com/StarRocks/starrocks/pull/46886) reappeared in version 3.3.3

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java:1]: What situation will reappear this problem?

In my test case, after bump hudi's version, this problem looks like solved.

Comment (plotor) [reply fe/fe-core/src/main/java/org/apache/hadoop/hbase/io/FSDataInputStreamWrapper.java:1]: I created a HUDI Catalog as follows:

```text
MySQL [doris_hudi]> show create catalog hudi_metastore_catalog \G;
*************************** 1. row ***************************
       Catalog: hudi_metastore_catalog
Create Catalog: CREATE EXTERNAL CATALOG `hudi_metastore_catalog`
PROPERTIES ("type"  =  "hudi",
"hive.metastore.uris"  =  "thrift://192.168.10.3:9083"
)
1 row in set (0.001 sec)
```

Then execute the following query:

```text
MySQL [doris_hudi]> select * from hudi_cow_nonpcf_tbl_1;
ERROR 1064 (HY000): Failed to get remote files, msg: com.google.common.util.concurrent.ExecutionError: com.google.common.util.concurrent.ExecutionError: java.lang.NoSuchMethodError: org.apache.hadoop.hdfs.client.HdfsDataInputStream.getReadStatistics()Lorg/apache/hadoop/hdfs/DFSInputStream$ReadStatistics;
```

The version of hudi and hbase in starrocks package is:

- hudi: 0.15.0
- hbase: 2.4.13

**Pull request merged**